openpype/plugins/publish/extract_color_transcode.py

```diff
@@ -0,0 +1,271 @@
+    This dict contains source colorspace information, collected by hosts.
+
+    Target colorspace is selected by profiles in the Settings, based on:
+    - families
+    - host
+    - task types
+    - task names
+    - subset names
+    """
+
+    label = "Transcode color spaces"
+    order = pyblish.api.ExtractorOrder + 0.019
+
+    optional = True
+
+    # Supported extensions
+    supported_exts = ["exr", "jpg", "jpeg", "png", "dpx"]
+
+    # Configurable by Settings
+    profiles = None
+    options = None
+
+    def process(self, instance):
+        if not self.profiles:
+            self.log.debug("No profiles present for color transcode")
+            return
+
+        if "representations" not in instance.data:
+            self.log.debug("No representations, skipping.")
+            return
+
+        if not is_oiio_supported():
+            self.log.warning("OIIO not supported, no transcoding possible.")
+            return
+
+        profile = self._get_profile(instance)
+        if not profile:
+            return
+
+        repres = instance.data.get("representations") or []
+        for idx, repre in enumerate(list(repres)):
+            self.log.debug("repre ({}): `{}`".format(idx + 1, repre["name"]))
+            if not self._repre_is_valid(repre):
+                continue
+
+            added_representations = False
+
+            colorspace_data = repre["colorspaceData"]
+            source_colorspace = colorspace_data["colorspace"]
+            config_path = colorspace_data.get("config", {}).get("path")
+            if not config_path or not os.path.exists(config_path):
+                self.log.warning("Config file doesn't exist, skipping")
+                continue
+
+            for _, output_def in profile.get("outputs", {}).items():
+                new_repre = copy.deepcopy(repre)
+
+                original_staging_dir = new_repre["stagingDir"]
+                new_staging_dir = get_transcode_temp_directory()
+                new_repre["stagingDir"] = new_staging_dir
+                files_to_convert = new_repre["files"]
+                if not isinstance(files_to_convert, list):
+                    files_to_convert = [files_to_convert]
+
+                files_to_delete = copy.deepcopy(files_to_convert)
+
+                output_extension = output_def["extension"]
+                output_extension = output_extension.replace('.', '')
+                if output_extension:
+                    if new_repre["name"] == new_repre["ext"]:
+                        new_repre["name"] = output_extension
+                    new_repre["ext"] = output_extension
+
+                    renamed_files = []
+                    _, orig_ext = os.path.splitext(files_to_convert[0])
+                    for file_name in files_to_convert:
+                        file_name = file_name.replace(orig_ext,
+                                                      "."+output_extension)
```

Comment:
Let me know if I'm wrong, but looks like this will generate a list of file name in `files_to_convert` of the output extension, which is later used as input path.
This means the input path files wont exist?

Reply:
Cleaned up.